program/src/processor.rs

```diff
@@ -249,7 +250,7 @@ fn check_vote_account(vote_account_info: &AccountInfo) -> Result<(), ProgramErro
         .ok_or(SinglePoolError::UnparseableVoteAccount)?;
 
     match u32::from_le_bytes(state_variant) {
-        1 | 2 => Ok(()),
+        1 | 2 | 3 => Ok(()),
         0 => Err(SinglePoolError::LegacyVoteAccount.into()),
         _ => Err(SinglePoolError::UnparseableVoteAccount.into()),
     }
```

Comment:
this gives us `VoteStateV4` support, it was necessary to do in this pr to get tests to pass. in the future it would be cleaner to switch to the `VoteState` parser though, which did not exist when this program was originally written. i didnt want to make any nontrivial functional changes in this one